states/wyoming/parse_pdf.py

```diff
@@ -0,0 +1,191 @@
+      elif 'Fax' in line:
+        key = 'fax'
+      elif 'Email' in line:
+        key = 'email'
+      else:
+        # Need to make sure following line doesn't have email suffix, too.
+        if (email_section and 
+            not any(suffix in line for suffix in email_suffixes) and
+            not any(suffix in row[i + 1] for suffix in email_suffixes)):
+          current_county = second_county
+          key = 'address'
+          email_section = False
+
+      if any(suffix in line for suffix in email_suffixes):
+        email_section = True
+
+      current_county[key].append(line)
+
+    formatted_rows.append(first_county)
+    formatted_rows.append(second_county)
+
+  # Remove last nonexistent county.
+  formatted_rows = formatted_rows[:-1]
+
+  return formatted_rows
+
+
+def generate_county_dict_list(formatted_rows):
+  counties = []
+
+  for formatted_row in formatted_rows:
+    county = defaultdict(list)
+    county['county'] = formatted_row['title'].strip(' \n')[:-len(' Clerk')]
+    county['locale'] = county['county']
+    county['address'] = ' '.join(formatted_row['address'])
+    # Remove space between '-' in zipcode.
+    if '-' in county['address']:
+      address_string = county['address']
+      dash_index = address_string.index('-')
+      if address_string[dash_index - 1] == ' ':
+        address_string = address_string[:dash_index - 1] + address_string[dash_index:]
+      if address_string[dash_index + 1] == ' ':
+        address_string = address_string[:dash_index] + address_string[dash_index + 1:]
+      county['address'] = address_string
+
+    full_email = ''
+    for email_line in formatted_row['email']:
+      if 'Email' in email_line:
+        continue
+      full_email += email_line
+      if any(suffix in email_line for suffix in email_suffixes):
+        county['emails'].append(full_email)
+        full_email = ''
+
+    full_phone_number = ''
+    for phone_number_line in formatted_row['phone_number']:
+      phone_number_line = phone_number_line.strip('Ph. ')
+      # One phone number as 'or' in it...
+      if 'or' in phone_number_line:
+        phone_number_line = phone_number_line[:phone_number_line.index('or')]
+      # Some phone numbers span multiple lines.
+      full_phone_number += phone_number_line
+    # Manually add '-' since some phone number span multiple lines.
+    full_phone_number = full_phone_number.replace('.', '')
+    full_phone_number = (full_phone_number[:3]
+                         + "-"
+                         + full_phone_number[3:6]
+                         + "-"
+                         + full_phone_number[6:])
+    county['phones'].append(full_phone_number.replace('.', '-'))
+
+    for fax_line in formatted_row['fax']:
+      fax_line = fax_line.strip('Fax ')
+      county['faxes'].append(fax_line.replace('.', '-'))
+      
+    counties.append(county)
+
+  return counties
+
+
+def main():
+  with open(dir_path(__file__) + '/results/wyoming_contact.pdf', 'rb') as fh:
```

Comment:
Can you download this file as your first step?  You should be able to directly download this using cache_request (perhaps you need to tweak cache_request to support binary formats as well).